### PR TITLE
Issue #185 make it possible to add item attributes as key/value pairs, not just values

### PR DESCRIPTION
**Pull request by el0ny**

### CODE REVIEW

File: pytest_reportportal/service.py [hunk 1]
```diff
@@ -630,12 +630,20 @@ def get_marker_value(item, keyword):
...
+        raw_attributes = [get_marker_value(item, k)
                       for k in item.keywords if get_marker(k) is not None
                       and k not in self.ignored_attributes]
 
-        attributes.extend([{"value": tag} for tag in
-                           item.session.config.getini('rp_tests_attributes')])
+        raw_attributes.extend([tag for tag in item.session.config.getini('rp_tests_attributes')])
+
+        attributes = []
+        for attr in raw_attributes:
```
Comment (iivanou) [inline pytest_reportportal/service.py:640]: Duplicates existing code in the get_launch_attributes function.

Comment (el0ny) [reply pytest_reportportal/service.py:643]: Duplicated code was moved to a new static function

**Comment by el0ny** [comment]
> Unit tests should pass. Remove duplicate code.

Now all tests pass correctly. All duplicates were resolved

File: pytest_reportportal/plugin.py [hunk 2]
```diff
@@ -49,12 +49,7 @@ def get_launch_attributes(rp_launch_attributes):
     """
     launch_attrs = []
     for rp_attr in rp_launch_attributes:
-        try:
-            key, value = rp_attr.split(':')
-            attr_dict = {'key': key, 'value': value}
-        except ValueError:
-            attr_dict = {'value': rp_attr}
-
+        attr_dict = PyTestServiceClass._get_attr_dict(rp_attr)
```
Comment (iivanou) [inline pytest_reportportal/plugin.py:52]: Why do not use re-use that function instead of modification?

File: pytest_reportportal/service.py [hunk 3]
```diff
@@ -630,14 +630,33 @@ def get_marker_value(item, keyword):
             get_marker = getattr(item, "get_closest_marker")
         except AttributeError:
             get_marker = getattr(item, "get_marker")
-        attributes = [{"value": get_marker_value(item, k)}
-                      for k in item.keywords if get_marker(k) is not None
-                      and k not in self.ignored_attributes]
-
-        attributes.extend([{"value": tag} for tag in
-                           item.session.config.getini('rp_tests_attributes')])
+        raw_attributes = [
```
Comment (iivanou) [inline pytest_reportportal/service.py:633]: I'd split the functionality of that method into multiple methods or functions.

File: pytest_reportportal/plugin.py (hunk 2 above)
Comment (el0ny) [reply pytest_reportportal/plugin.py:52]: What do you mean? You want me to call get_launch_attributes in _get_marker_value? I don't think that is a good idea. First, I'm not sure if I can call the whole get_launch_attributes function. Second, there is a problem of accessing plugin from service module. In fact right now there is no such functions in plugin that are being called from service. On the other hand whole PyTestServiceClass is being imported in plugin module.

File: pytest_reportportal/service.py (hunk 3 above)
Comment (el0ny) [reply pytest_reportportal/service.py]: I have created a new get_raw_attr function inside _get_item_markers to contain some of it's functionality

File: pytest_reportportal/plugin.py (hunk 2 above)
Comment (el0ny) [reply pytest_reportportal/plugin.py:52]: What I mean is that importing get_launch_attributes into service module causes cross import error

**Pull request closed without merging**